### PR TITLE
Performing Always and Never Sample Steps

### PR DESCRIPTION
**Pull request by bobstrecansky**
Creating a PR in order to include `Always Sample` and `Never Sample` samplers, as per the SDK spec:
This is a very rudimentry, blunt version of : https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-tracing.md#sampler

Performing this for now with the intention of updating it later to more closely follow the spec so work on the exporter can begin.

It makes sense to me to add the probability sampler after this is more explicitly defined:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-tracing.md#probability-sampler-algorithm

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +0,0 @@
-FROM php:7.2-buster
```
Comment (bobstrecansky) [inline Dockerfile]: This Dockerfile was vestigial; we are coordinating them all in the `resources` folder now.

File: examples/AlwaysOffTraceExample.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php
+require __DIR__.'/../vendor/autoload.php';
+use OpenTelemetry\Tracing\Builder;
+use OpenTelemetry\Tracing\SpanContext;
+use OpenTelemetry\Tracing\Sampler\AlwaysOffSampler;
+$sampler = AlwaysOffSampler::shouldSample();
+if($sampler) {
+    $spanContext = SpanContext::generate(); // or extract from headers
+    $tracer = Builder::create()->setSpanContext($spanContext)->getTracer();
```
Comment (AustinSchoen) [inline examples/AlwaysOffTraceExample.php]: Was the builder removed in another PR?

Comment (bobstrecansky) [reply examples/AlwaysOffTraceExample.php]: Yup; this gets removed in the next PR:
https://github.com/open-telemetry/opentelemetry-php/pull/31

File: examples/AlwaysOnTraceExample.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?php
+require __DIR__.'/../vendor/autoload.php';
+use OpenTelemetry\Tracing\Builder;
+use OpenTelemetry\Tracing\SpanContext;
+use OpenTelemetry\Tracing\Sampler\AlwaysOnSampler;
+$sampler = AlwaysOnSampler::shouldSample();
+if($sampler) {
+    $spanContext = SpanContext::generate(); // or extract from headers
+    $tracer = Builder::create()->setSpanContext($spanContext)->getTracer();
```
Comment (bobstrecansky) [inline examples/AlwaysOnTraceExample.php]: We should change this after the `Builder` goes away in the subsequent PR

**Pull request merged**